jigglypuffRL/common/actor_critic.py

```diff
@@ -69,15 +77,74 @@ def forward(self):
         raise NotImplementedError
 
 
+# Inherited classes
 class ActorCritic(nn.Module):
-    def __init__(self, env, Actor, Critic):
-        super(ActorCritic).__init__(env)
+    def __init__(self, env, network_type, n_hidden, noise_std=0, det_stoc="det"):
+        super(ActorCritic, self).__init__()
 
-        self.actor = Actor(env)
-        self.critic = Critic(env)
+        self.actor, self.critic = get_actor_critic_from_name(network_type)
+        self.actor = self.actor(env, n_hidden)
+        self.critic = self.critic(env, n_hidden)
+        self.det_stoc = det_stoc
+        self.noise_std = noise_std
 
-    def value(self, x):
-        return self.critic.forward(x)
+    def value(self, s, a):
+        return self.critic.forward(s, a)
 
-    def sample_action(self, x):
-        return self.actor.sample_action(x)
+    def select_action(self, x):
+        with torch.no_grad():
+            if self.det_stoc == "det":
+                return self.actor(x).cpu().numpy()
+            elif self.det_stoc == "stoc":
+                return self.actor.sample_action(x)
+            else:
+                raise ValueError
+
+
+class MlpActor(BaseActor):
+    def __init__(self, env, n_hidden):
+        super(MlpActor, self).__init__(env)
+
+        self.action_limit = env.action_space.high[0]
+
+        self.fc1 = nn.Linear(self.state_dim, n_hidden)
+        self.fc2 = nn.Linear(n_hidden, n_hidden)
+        self.fc3 = nn.Linear(n_hidden, self.action_dim)
+
+    def forward(self, x):
+        model = nn.Sequential(self.fc1, nn.ReLU(), self.fc2, nn.ReLU(), self.fc3)
```

Comment:
We should definitely shift this definition to the constructor.  And have it as one of the class variables rather than a temporary object, since this is literally the core of `MlpActor`